game-app/game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbForumPoster.java

```diff
@@ -157,12 +168,18 @@ private String uploadSaveGame(
       if (status == HttpURLConnection.HTTP_OK) {
         final String json = EntityUtils.toString(response.getEntity());
         try {
-          final String url = (String) YamlReader.readMap(json).get("url");
-          return "\n[Savegame](" + url + ")";
+          final Map<String, Object> jsonObject = YamlReader.readMap(json);
+          final Map<?, ?> responseObject = (Map<?, ?>) jsonObject.get("response");
+          // This is a temporary hack to handle old versions of nodeBB forum json
+          if (responseObject == null) {
+            return (String) jsonObject.get("url");
+          }
+          final List<?> images = (List<?>) Preconditions.checkNotNull(responseObject.get("images"));
+          final Map<?, ?> imageObject = (Map<?, ?>) images.get(0);
+          return (String) Preconditions.checkNotNull(imageObject.get("url"));
         } catch (final Exception e) {
           // This is a temporary hack to handle old versions of nodeBB forum json
```

Comment:
Is this comment still current, any idea? Did this behavior get tested, or is just seemingly the right fix (but not actually validated?) This logic down here also helps add to the idea of adding testing so we can validate that we actually handle old & new formats.

Is the 'old versions' of nodeBB largely just the TripleA forum now?

Reply:
> Is the 'old versions' of nodeBB largely just the TripleA forum now?

Yes, it needs to be updated, but I haven't gotten around to do so. (Also NodeBB 2.0 has been released and I'm not sure if there are any breaking changes that need any special treatment.)

I verified the code works with the a&a forum and our forum and I believe the branch where I check if the "url" key exists will no longer be executed (haven't checked though)